spring-boot/admin/src/test/java/eu/coatrack/admin/controllers/PublicAPIAuthenticationAndAuthorizationViaHeaderTokenTest.java

```diff
@@ -0,0 +1,66 @@
+package eu.coatrack.admin.controllers;
+
+import eu.coatrack.admin.security.CustomUserInfoTokenServices;
+import eu.coatrack.api.ServiceApiDTO;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class PublicAPIAuthenticationAndAuthorizationViaHeaderTokenTest {
+
+    private static final String AUTHORIZATION_TOKEN = "testToken";
+    private static final String PUBLIC_API_URL = "http://localhost:8080/public-api/services";
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @MockBean
+    private CustomUserInfoTokenServices customUserInfoTokenServicesMock;
+
+    //test to call the public api and get the services
+    @Test
+    public void CallThePublicAPIWithAuthorizationBearerToken_ShouldRetrieveListOfServices() {
+
+        UsernamePasswordAuthenticationToken userAuthentication = new UsernamePasswordAuthenticationToken("exampleCompanyInc",
+                "test", Collections.singleton(new SimpleGrantedAuthority("ROLE_USER")));
+        OAuth2Request request = new OAuth2Request(null, null, null, true, null, null, null, null, null);
+        when(customUserInfoTokenServicesMock.loadAuthentication(AUTHORIZATION_TOKEN)).thenAnswer(invocation -> new OAuth2Authentication(request, userAuthentication));
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBearerAuth(AUTHORIZATION_TOKEN);
+        HttpEntity httpRequest = new HttpEntity(headers);
+
+        ResponseEntity<List<ServiceApiDTO>> getListOfServicesViaPublicAPI = restTemplate.exchange(PUBLIC_API_URL, HttpMethod.GET,
+                httpRequest, new ParameterizedTypeReference<List<ServiceApiDTO>>() {
+                });
+
+        Assertions.assertTrue(getListOfServicesViaPublicAPI.getBody().stream().allMatch(serviceOwner -> serviceOwner.getServiceOwnerUsername().equals("exampleCompanyInc")));
```

Comment:
* I'd remove "Assertions.". 
* Also a lambda expression as a function argument is hard to read, please create an extra variable for that. 
* The definition of allMatch() states: "If the stream is empty then true is returned and the predicate is not evaluated.". Therefore it might happen the stream is empty, which indicates an error, but the test still passes. False-positive tests are really hard to detect. They should be avoided at any cost.

My suggestion:
* Collect all service (maybe using the "filter()" function), e.g., to a `List<ServiceApiDTO> serviceOwnerList` variable.
* `assertEquals (1, serviceOwnerList.size())`

Reply:
First of all, I really liked this suggestion, it improved a lot the test code ;)

Regarding the filter, what will happen is that all the services belong to the same user, so I will get 3 at the end, as these are the services we have on the DatabaseInitializer.
Based on this, instead of a `assertEquals (1, serviceOwnerList.size())`, I did a `<List<ServiceApiDTO>> getListOfServicesViaPublicAPI` size compared to the filtered list size, so they should have the same size

Reply:
This is Done